python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py

```diff
@@ -126,8 +128,10 @@ class Metadata(Node):
     personal_sensitive_information: str | None = None
 
     def __post_init__(self):
-        """Checks arguments of the node."""
+        """Checks arguments of the node and setup ID."""
         # Define parents.
+        if not self.id:
+            self.id = generate_uuid()
```

Comment:
Rollback. This should happen in base_node.

Reply:
Mmm I thought we wanted to have an explicitly set `id` for nodes other than Metadata (i.e. Field, FileObject, FileSet, RecordSet) -- entweder an `id` or derive that from `name` (eventually).

For `Metadata` is different, given that the id is only used under the water, so I generate it randomly. Isn't it so?

Reply:
You should define `id` only in the `Node` class that they all inherit from:

```
class Node:
  id = dataclasses.field(default_factory=generate_uuid)
```

Reply:
But then it would always have a default value. I thought we wanted to avoid setting a default value on ids for specific  nodes?

Reply:
I think every node needs a UUID

Reply:
Done.